src/Raven.Server/Documents/PeriodicBackup/RavenFtpClient.cs

```diff
@@ -130,72 +143,126 @@ private void ExtractUrlAndDirectories(out string url, out List<string> dirs)
             var uri = new Uri(_url);
 
             dirs = uri.AbsolutePath.TrimStart('/').TrimEnd('/').Split("/").ToList();
-            var port = _port ?? (uri.Port > 0 ? uri.Port : DefaultFtpPort);
-            if (port < 1 || port > 65535)
-                throw new ArgumentException("Port number range: 1-65535");
 
-            url = $"{uri.Scheme}://{uri.Host}:{port}";
+            url = $"{uri.Scheme}://{uri.Host}";
         }
 
-        private FtpWebRequest CreateFtpWebRequest(string url, string method, bool keepAlive)
+        internal FtpClient CreateFtpClient(string url, bool keepAlive)
         {
-#pragma warning disable SYSLIB0014
-            var request = (FtpWebRequest)WebRequest.Create(new Uri(url));
-#pragma warning restore SYSLIB0014
-            request.Method = method;
-
-            request.Credentials = new NetworkCredential(_userName, _password);
-            request.EnableSsl = _useSsl;
+            var client = new FtpClient(url);
+            client.Config.ConnectTimeout = -1;
```

Comment:
this can hang now forever, what better value we can put here?

Reply:
Done